docs/en/migration/model.md

```diff
@@ -1,3 +1,477 @@
+                    log_vars=dict(
+                        loss1=loss1.sum().item(),
+                        loss2=loss2.sum().item()))
+
+    def train_step(self, data, optimizer=None):
+        return self(*data, return_loss=True)
+
+    def val_step(self, data, optimizer=None):
+        return self(*data, return_loss=False)
+```
+
+</div>
+  </td>
+  <td valign="top" class='two-column-table-wrapper'><div style="overflow-x: auto">
+
+```python
+class MMEngineToyModel(BaseModel):
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.linear = nn.Linear(1, 1)
+
+    def forward(self, img, label, mode):
+        if mode == 'loss':
+            feat = self.linear(img)
+            loss1 = (feat - label).pow(2)
+            loss2 = (feat - label).abs()
+            return dict(loss1=loss1, loss2=loss2)
+        elif mode == 'predict':
+            return [_feat for _feat in feat]
+        else:
+            pass
+
+    # The equivalent code snippet of `train_step`
+    # def train_step(self, data, optim_wrapper):
+    #     data = self.data_preprocessor(data)
+    #     loss_dict = self(*data, mode='loss')
+    #     loss_dict['loss1'] = loss_dict['loss1'].sum()
+    #     loss_dict['loss2'] = loss_dict['loss2'].sum()
+    #     loss = (loss_dict['loss1'] + loss_dict['loss2']).sum()
+    #     Call the optimizer wrapper to update parameters.
+    #     optim_wrapper.update_params(loss)
+    #     return loss_dict
+```
+
+</td>
+</div>
+</tr>
+</thead>
+</table>
+
+```{note}
+See more information about `data_preprocessor` and `optim_wrapper` in docs [optim_wrapper](../tutorials/optim_wrapper.md) and [data_preprocessor](../tutorials/model.md).
+```
+
+The main differences of model in MMCV and MMEngine can be summarized as follows:
+
+- `MMCVToyModel` inherits from `nn.Module`, and `MMEngineToyModel` inherits from `BaseModel`
+
+- `MMCVToyModel` must implement `train_step` methods and return a `dict` with keys `loss`, `log_vars`, and `num_samples`. `MMEngineToyModel` only needs to implement `forward` method for high level tasks, and return a differentiable loss `dict`.
+
+- `MMCVToyModel.forward` and `MMEngineToyModel.forward` must match with `train_step` which will call it. Since `MMEngineToyModel`does not override the `train_step` will `BaseModel.train_step` will be directly called, which requires that forward must accept `mode` parameter. Find more details in [tutorials of model](../tutorials/model.md)
+
+### Custom optimization process
+
+Takes training a GAN model as an example, generator and discriminator need to be optimized in turn and the optimization strategy could change with the training iteration grows. Therefore it could be hard to use `OptimizerHook` to meet such requirement in MMCV. GAN model based on MMCV will accept optimizer in `train_step` and update parameters in it. Actually, MMEngine borrows this way and simplify it by passing an [optim_wrapper](../tutorials/optim_wrapper.md) rath than an optimizer.
```

Comment:
```suggestion
Takes training a GAN model as an example, generator and discriminator need to be optimized in turn and the optimization strategy could change as the training iteration grows. Therefore it could be hard to use `OptimizerHook` to meet such requirement in MMCV. GAN model based on MMCV will accept optimizer in `train_step` and update parameters in it. Actually, MMEngine borrows this way and simplify it by passing an [optim_wrapper](../tutorials/optim_wrapper.md) rather than an optimizer.
```